### PR TITLE
feat: disable star rating by default

### LINKED ISSUE #9782: Make star rating disabled by default
Customers using the star rates to show the product rates would not expect to explicitly set the disabled input to true. There's literally one place where we need the enable the rating (in the create review component).

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
closes #9782

### CODE REVIEW

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts [hunk 1]
```diff
@@ -1,57 +1,66 @@
...
    * Emits the given rating when the user clicks on a star.
    */
   // tslint:disable-next-line:no-output-native
   @Output() change = new EventEmitter<number>();
 
-  private initialRate = 0;
-
-  iconTypes = ICON_TYPE;
-
-  constructor(protected el: ElementRef, protected renderer: Renderer2) {}
```
Comment (znikola) [inline projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts:41]: Removing this ctor is a breaking change. Would like to address it in this ticket, or create a new one?

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts [hunk 2]
```diff
@@ -1,57 +1,66 @@
...
-
-  ngOnInit(): void {
```
Comment (znikola) [inline projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts:43]: Also a breaking change

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts [hunk 3]
```diff
@@ -1,57 +1,66 @@
...
-    this.setRate(this.rating, true);
+  setRate(value: number): void {
+    if (this.disabled) {
+      return;
+    }
+    this.rating = value;
   }
 
-  setRate(value: number, force?: boolean): void {
```
Comment (znikola) [inline projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts:47]: Also a breaking change, as the second parameter was removed.

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts [hunk 4]
```diff
@@ -62,11 +71,4 @@ export class StarRatingComponent implements OnInit {
     this.setRate(rating);
     this.change.emit(rating);
   }
-
-  setRateOnEvent(event: any, rating: number) {
```
Comment (znikola) [inline projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts:66]: This is a breaking change as well

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.html [hunk 5]
```diff
@@ -1,10 +1,9 @@
 <cx-icon
   *ngFor="let i of [1, 2, 3, 4, 5]"
-  [type]="iconTypes.STAR"
+  [type]="icon"
   class="star"
   (mouseover)="setRate(i)"
-  (mouseout)="setRate(0)"
-  (keydown)="setRateOnEvent($event, i)"
```
Comment (znikola) [inline projects/storefrontlib/src/shared/components/star-rating/star-rating.component.html:7]: I think that removing these events is considered as a breaking change

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.spec.ts [hunk 6]
```diff
@@ -11,45 +12,149 @@ class MockIconComponent {
...
+  });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(StarRatingComponent);
-    starRatingComponent = fixture.componentInstance;
+    component = fixture.componentInstance;
+  });
+
+  it('should be disabled by default', () => {
+    expect(component.disabled).toBeTruthy();
```
Comment (znikola) [inline projects/storefrontlib/src/shared/components/star-rating/star-rating.component.spec.ts]: Maybe we want to nest this `it` block under the `describe('disabled = true (default)')`?

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.spec.ts [hunk 7]
```diff
@@ -11,45 +12,149 @@ class MockIconComponent {
...
+      expect(component.rating).toEqual(0);
+      component.setRate(3);
+      expect(component.rating).toEqual(0);
+    });
+
+    it('should not save rate', () => {
+      expect(component.rating).toEqual(0);
+      component.saveRate(3);
+      expect(component.rating).toEqual(0);
+    });
```
Comment (znikola) [inline projects/storefrontlib/src/shared/components/star-rating/star-rating.component.spec.ts:61]: This seems to test the same thing as the test above it. Was this intentional?

**Review by znikola — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.spec.ts (hunk 7 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/shared/components/star-rating/star-rating.component.spec.ts:61]: slightly different, set vs save.

Comment (znikola) [reply projects/storefrontlib/src/shared/components/star-rating/star-rating.component.spec.ts:61]: Right, I didn't catch it 👍

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.html (hunk 5 above)
Comment (znikola) [reply projects/storefrontlib/src/shared/components/star-rating/star-rating.component.html:7]: Our migration schematics support inserting an html comment if customers are calling these methods. 
You can leverage it in `projects/schematics/src/migrations/3_0/component-deprecations/component-deprecations.ts`

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts (hunk 2 above)
Comment (znikola) [reply projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts:43]: We can use `projects/schematics/src/migrations/3_0/methods-and-properties-deprecations/methods-and-properties-deprecations.ts` to notify customers about this deprecation via a comment:

```ts
{
    class: 'StarRatingComponent',
    importPath: SPARTACUS_STOREFRONTLIB,
    deprecatedNode: `ngOnInit`,
    comment: `// ${TODO_SPARTACUS} ...`,
  },
```

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts (hunk 4 above)
Comment (znikola) [reply projects/storefrontlib/src/shared/components/star-rating/star-rating.component.ts:66]: Similar to https://github.com/SAP/spartacus/pull/9783/files#r523978462, we can leverage update schematics mechanism for this

**Review by znikola — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/shared/components/star-rating/star-rating.component.html (hunk 5 above)
Comment (znikola) [reply projects/storefrontlib/src/shared/components/star-rating/star-rating.component.html:7]: E.g.

```ts
  selector: 'cx-star-rating',
  componentClassName: 'StarRatingComponent',
  removedProperties: [
    {
      name: 'setRateOnEvent',
      comment: `'setRateOnEvent' output was removed....`,
    },
  ],
```

**Review by znikola — APPROVED** [review]

**Pull request merged**